### PR TITLE
Homework 6

### PR DESCRIPTION
**Pull request by KTsarenko**

### CODE REVIEW

File: homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                this.speed = this.setSpeed();
+            }, 1000);
+            const timerRun = setInterval(() => {
+                console.log(`name: ${this.name} breed: ${this.breed} distance: ${this.distance} `);
+            }, 2000);
+            setTimeout(() => {
+                clearInterval(timerRun);
+                clearInterval(race);
+                resolve();
+            }, 10000);
```
Comment (markoffden) [inline homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js]: `run()` should not include race related functionality, logically it's wrong, `run()` should only make horse run

File: homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    startRace() {
+        const timeRun = 10;
+        this.horses.forEach((horse, i) => {
+            const result = horse.run(timeRun);
+            result.then(() => {
+                if (i === this.horses.length - 1) {
+                    const winner = this.horses.reduce((acc, current) => {
+                        const winnerHorse = acc.distance > current.distance ? acc : current;
+                        return winnerHorse;
```
Comment (markoffden) [inline homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js]: you could use built in JS sorting functionality here

File: homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js (hunk 1 above)
Comment (KTsarenko) [reply homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js]: removed

File: homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js (hunk 2 above)
Comment (KTsarenko) [reply homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js]: this part has been changed

File: homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    constructor(horsesNames) {
+        this.horses = this.createRace(horsesNames);
+    }
+
+    createRace(horsesArr) {
+        this.racerArr = [];
+        horsesArr.forEach((horse) => {
+            const racer = new Racer(horse.name, horse.breed);
+            this.racerArr.push(racer);
+        });
```
Comment (markoffden) [inline homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js]: this is a perfect case for `map()`, cause you create array out of array

Comment (KTsarenko) [reply homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js]: replaced

**Pull request merged**